.github/workflows/ci.yml

```diff
@@ -87,18 +128,31 @@ jobs:
         java-version: 11
 
     - name: install linux dependencies
-      if: ${{ matrix.host == 'ubuntu-latest' }}
+      if: ${{ matrix.host == 'ubuntu-latest'}}
       run: sudo apt-get update && sudo apt-get install libwebkit2gtk-4.0-dev libgtk-3-dev squashfs-tools
 
+    - name: install dependencies for i686
+      if: ${{ matrix.host == 'ubuntu-latest'}}
+      run: sudo dpkg --add-architecture i386 && sudo apt-get update && sudo apt-get install libwebkit2gtk-4.0-dev:i386 libgtk-3-dev:i386 squashfs-tools:i386 libssl3:i386 gcc-multilib libglib2.0-0:i386
+
     - name: create project
+      if: ${{ !((matrix.host == 'macos-latest' || matrix.host == 'windows-latest' ) && matrix.host-arch == 'x86')}}
       run: ./x new template
-
+      
     # only linux can target linux as it is not clear what that even is exactly
     # glibc/musl gtk/qt x11/wayland
     # windows doesn't support posix symlinks so can't cross compile to macos/ios
     - name: build project
       if: >
-        !(matrix.host == 'macos-latest' && matrix.target.platform == 'linux' ||
-          matrix.host == 'windows-latest' && contains(fromJson('["linux", "macos", "ios"]'), matrix.target.platform))
+        ${{
+          !(matrix.host-arch == 'x64' && matrix.target.arch == 'x86' ||
+            matrix.host == 'macos-latest' && matrix.host-arch == 'x86' ||
+            matrix.host == 'windows-latest' && matrix.host-arch == 'x86' ||
+            matrix.host == 'macos-latest' && matrix.target.platform == 'linux' ||
+            matrix.host == 'windows-latest' && contains(fromJson('["linux", "macos", "ios"]'), matrix.target.platform)
+          )
+        }}    
```

Comment:
This thing is getting out of hand, perhaps we can create a separate `job:` for it with a `matrix` that uses `exclude`/`include` instead.